feature/splash/src/main/java/com/epmedu/animeal/splash/presentation/SplashViewModel.kt

```diff
@@ -1,28 +1,64 @@
 package com.epmedu.animeal.splash.presentation
 
+import android.util.Log
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.amplifyframework.auth.AuthSession
+import com.epmedu.animeal.auth.AuthRequestHandler
 import com.epmedu.animeal.common.presentation.viewmodel.delegate.DefaultEventDelegate
 import com.epmedu.animeal.common.presentation.viewmodel.delegate.EventDelegate
-import com.epmedu.animeal.profile.data.repository.ProfileRepository
+import com.epmedu.animeal.splash.domain.FetchUserSessionUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class SplashViewModel @Inject constructor(
-    private val profileRepository: ProfileRepository
-) : ViewModel(),
+    private val fetchUserSessionUseCase: FetchUserSessionUseCase,
+) :
+    ViewModel(),
     EventDelegate<SplashViewModel.Event> by DefaultEventDelegate() {
 
-    fun verifyProfileSaved() {
-        viewModelScope.launch {
-            if (profileRepository.isProfileSaved()) {
-                sendEvent(Event.NavigateToHome)
+    private val sessionRequestHandler = object : AuthRequestHandler {
+        override fun onSuccess(result: Any?) {
+            Log.d("sessionRequest", "result -> $result")
+            if (result is AuthSession) {
+                processSession(result)
             } else {
-                sendEvent(Event.NavigateToOnboarding)
+                navigateToOnboarding()
             }
         }
+
+        override fun onError(exception: Exception) {
+            Log.d("sessionRequest", "exception -> $exception")
+            navigateToOnboarding()
+        }
+    }
+
+    fun verifyProfileSaved() {
+        viewModelScope.launch {
+            fetchUserSessionUseCase(sessionRequestHandler)
+        }
+    }
+
+    fun processSession(session: AuthSession) {
+        if (session.isSignedIn) {
+            navigateToHome()
+        } else {
+            navigateToOnboarding()
+        }
+    }
+
+    private fun navigateToHome() {
+        viewModelScope.launch {
+            sendEvent(Event.NavigateToHome)
+        }
+    }
+
+    private fun navigateToOnboarding() {
+        viewModelScope.launch {
+            sendEvent(Event.NavigateToOnboarding)
+        }
     }
 
     sealed interface Event {
```

Comment:
Because it is being used in SplashScreen: What do you think about extracting it to its own file and rename it, like 'SplashScreenEvents' or something?

Reply:
Good idea, done, please have a look